src/state/queries/messages/conversation.ts

```diff
@@ -46,6 +49,44 @@ export function useMarkAsReadMutation() {
         {
           convoId,
           messageId,
+          unreadCount: 0, // TODO: figure out which field convo API uses
+        },
+        {
+          encoding: 'application/json',
+          headers: DM_SERVICE_HEADERS,
+        },
+      )
+    },
+    onMutate({convoId}) {
+      if (!convoId) throw new Error('No convoId provided')
+      optimisticUpdate(convoId)
+    },
+    onSettled() {
+      queryClient.invalidateQueries({queryKey: LIST_CONVOS_KEY})
+    },
+  })
+}
+
+export function useMarkAsUnreadMutation() {
+  const optimisticUpdate = useOnMarkAsUnread()
+  const queryClient = useQueryClient()
+  const agent = useAgent()
+
+  return useMutation({
+    mutationFn: async ({
+      convoId,
+      messageId,
+    }: {
+      convoId: string
+      messageId?: string
+    }) => {
+      if (!convoId) throw new Error('No convoId provided')
+
+      await agent.api.chat.bsky.convo.updateRead(
+        {
+          convoId,
+          messageId,
+          unreadCount: 1, // TODO: figure out which field convo API uses
```

Comment:
@haileyok @mozzius any of you know if this `api.chat.bsky.convo.updateRead` takes a count input?

This is the current type def `ChatBskyConvoUpdateRead.InputSchema`:

```ts
export interface InputSchema {
  convoId: string
  messageId?: string
  [k: string]: unknown
}
```

Reply:
No, it does not take an count input

Reply:
@mary-ext any insight on how to persist this unread count? the optimistic update only last for a couple of seconds

https://github.com/user-attachments/assets/9d1ffba9-0795-4024-b1f3-e595c9814950